snmp/datadog_checks/snmp/snmp.py

```diff
@@ -50,6 +101,7 @@ class SnmpCheck(AgentCheck):
     _executor = None
     _NON_REPEATERS = 0
     _MAX_REPETITIONS = 25
+    _thread_factory = threading.Thread
```

Comment:
Perhaps a small comment to express the intent here? (at first I was surprised we didn't just call `threading.Thread()` in the check body)

```suggestion
    _thread_factory = threading.Thread  # Store as an attribute for easier mocking.
```